src/Aspire.Dashboard/Otlp/Http/OtlpHttpEndpointsBuilder.cs

```diff
@@ -0,0 +1,205 @@
+        {
+            if (request.Message == null)
+            {
+                return Results.Empty;
+            }
+            return OtlpResult.Response(service.Export(request.Message));
+        });
+    }
+
+    private enum KnownContentType
+    {
+        None,
+        Protobuf,
+        Json
+    }
+
+    private static KnownContentType GetKnownContentType(string? contentType, out StringSegment charSet)
+    {
+        if (contentType != null && MediaTypeHeaderValue.TryParse(contentType, out var mt))
+        {
+            if (string.Equals(mt.MediaType, JsonContentType, StringComparison.OrdinalIgnoreCase))
+            {
+                charSet = mt.CharSet;
+                return KnownContentType.Json;
+            }
+
+            if (string.Equals(mt.MediaType, ProtobufContentType, StringComparison.OrdinalIgnoreCase))
+            {
+                charSet = mt.CharSet;
+                return KnownContentType.Protobuf;
+            }
+        }
+
+        charSet = default;
+        return KnownContentType.None;
+    }
+
+    private static T AddOtlpHttpMetadata<T>(this T builder) where T : IEndpointConventionBuilder
+    {
+        builder
+            .RequireAuthorization(OtlpAuthorization.PolicyName)
+            .Add(b => b.Metadata.Add(new SkipStatusCodePagesAttribute()));
+        return builder;
+    }
+
+    private sealed class MessageBindable<TMessage> : IBindableFromHttpContext<MessageBindable<TMessage>> where TMessage : IMessage<TMessage>, new()
+    {
+        public static readonly MessageBindable<TMessage> Empty = new MessageBindable<TMessage>();
+
+        public TMessage? Message { get; private set; }
+
+        public static async ValueTask<MessageBindable<TMessage>?> BindAsync(HttpContext context, ParameterInfo parameter)
+        {
+            switch (GetKnownContentType(context.Request.ContentType, out var charSet))
+            {
+                case KnownContentType.Protobuf:
+                    try
+                    {
+                        var message = await ReadOtlpData(context, data =>
+                        {
+                            var message = new TMessage();
+                            message.MergeFrom(data);
+                            return message;
+                        }).ConfigureAwait(false);
+
+                        return new MessageBindable<TMessage> { Message = message };
+                    }
+                    catch (BadHttpRequestException ex)
+                    {
+                        context.Response.StatusCode = ex.StatusCode;
+                        return Empty;
+                    }
+                case KnownContentType.Json:
+                default:
+                    context.Response.StatusCode = StatusCodes.Status415UnsupportedMediaType;
+                    return Empty;
+            }
+        }
+    }
+
+    private sealed class OtlpResult<T> : IResult where T : IMessage
```

Comment:
Is it helpful for this result type to implement `IEndpointMetadataProvider` so that information about the response produced from these endpoints can surface to documentation tools like OpenAPI?

Reply:
Right now, the dashboard doesn't produce OpenAPI. This can improve later if that changes.